src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -242,15 +242,40 @@ pub(crate) fn is_ci_llvm_available(config: &Config, asserts: bool) -> bool {
 
 /// Returns true if we're running in CI with modified LLVM (and thus can't download it)
 pub(crate) fn is_ci_llvm_modified(config: &Config) -> bool {
-    CiEnv::is_rust_lang_managed_ci_job() && config.rust_info.is_managed_git_subrepository() && {
-        // We assume we have access to git, so it's okay to unconditionally pass
-        // `true` here.
-        let llvm_sha = detect_llvm_sha(config, true);
-        let head_sha =
-            output(helpers::git(Some(&config.src)).arg("rev-parse").arg("HEAD").as_command_mut());
-        let head_sha = head_sha.trim();
-        llvm_sha == head_sha
+    // If not running in a CI environment, return false.
+    if !CiEnv::is_ci() {
+        return false;
+    }
+
+    // In rust-lang/rust managed CI, assert the existence of the LLVM submodule.
+    if CiEnv::is_rust_lang_managed_ci_job() {
+        assert!(
+            config.in_tree_llvm_info.is_managed_git_subrepository(),
+            "LLVM submodule must be fetched in rust-lang/rust managed CI builders."
+        );
     }
+    // If LLVM submodule isn't present, skip the change check as it won't work.
+    else if !config.in_tree_llvm_info.is_managed_git_subrepository() {
+        return false;
+    }
+
+    let commit = t!(get_closest_merge_commit(Some(&config.src), &config.git_config(), &[]));
+
+    #[allow(clippy::let_and_return)] // makes it look more clear
+    let is_modified = !t!(helpers::git(Some(&config.src))
+        .args(["diff-index", "--quiet", &commit])
+        .arg("--")
+        .args([
+            config.src.join("src/llvm-project"),
```

Comment:
Could you please unify these three files with the same filelist used in `detect_llvm_sha`?

Reply:
It's no longer necessary.